_source/_posts/2024-01-24-spring-boot-authorization.md

```diff
@@ -0,0 +1,569 @@
+                 // exports.onContinuePostLogin = async (event, api) => {                                                 
+                 // };                                                                    
+```
+
+You can list the available actions with the following command:
+
+```shell
+auth0 actions list
+```
+The output will show the deployment status of each action:
+
+```text
+=== dev-avup2laz.us.auth0.com actions
+
+  ID                                    NAME       TYPE        STATUS  DEPLOYED  
+  da49ae42-b5e4-496a-8305-4fff437f813b  Add Roles  post-login  built   ✗         
+```
+
+Note the `DEPLOYED` status is `x`. Go ahead and deploy it using the action ID:
+
+```shell
+auth0 actions deploy \<ACTION_ID\>
+```
+
+Once the action is deployed, you must attach it to the login flow. You can do this with Auth0 [Management API for Actions](https://auth0.com/docs/api/management/v2#!/Actions/patch_bindings):
+
+```shell
+auth0 api patch "actions/triggers/post-login/bindings" \
+  --data '{"bindings":[{"ref":{"type":"action_id","value":"\<ACTION_ID\>"},"display_name":"Add Roles"}]}'
+```
+
+You can visualize the flow in the Auth0 dashboard. Sign in and on the left menu you choose **Actions**, then in the **Flows** screen, choose **Login**.
+
+{% img blog/spring-boot-authorization/login-flow.png alt:"Custom Auth0 Login Action" width:"600" %}{: .center-image }
+
+Now the role has been assigned and mapped to the ID and access tokens, and the UI will display the links to perform write operations. Still, the API server is not yet enforcing RBAC.
+
+### Enable RBAC in Auth0
+
+The next step is to enable RBAC for the API in Auth0. You can do it with the following Auth0 CLI commands:
+
+```shell
+auth0 apis list
+```
+
+Copy the Menu API ID and use it for the next command:
+
+```shell
+auth0 api patch "resource-servers/API_ID" \
+  --data '{ "enforce_policies": true, "token_dialect": "access_token_authz" }'
+```
+
+The `token_dialect` value `access_token_authz` enables the permissions to be included in the access token, in a custom `permissions` claim.
+
+Assign the Menu API permissions to the `menu-admin` role:
+
+```shell
+auth0 roles permissions add
+```
+
+Follow the instructions, and make sure to select all the API permissions:
+- `create:items`
+- `delete:items`
+- `update:items`
+
+Next, implement RBAC in the Spring Boot API.
+
+### Implement RBAC in the Spring Boot API
+
+The Okta Starter provides a simple way to specify the claim from which authorities must be extracted. In the `application.properties` file, add the following property:
+
+```property
+okta.oauth2.groupsClaim=permissions
+```
+With the `@PreAuthorize` annotation you define the required permission to perform the endpoint operation. The final controller implementation should look like this:
+
+```java
+// src/main/java/com/example/menu/web/ItemController.java
+package com.example.menu.web;
```

Comment:
lets skip the imports since its in the git repo anyway, else this code block will look huge in the blog